### ISSUE TITLE
akka-http_2.11 for 10.1.1 broken in jcenter

### ISSUE THREAD

**Issue by brharrington** [akka/akka-http#1972]
FYI, I'm seeing errors pulling in akka-http_2.11 10.1.1 from jcenter.

```
[warn] problem while downloading module descriptor: https://jcenter.bintray.com/com/typesafe/akka/akka-http_2.11/10.1.1/akka-http_2.11-10.1.1.pom: invalid sha1: expected=<?xml computed=7ab0cc491f566d7824d76c3f7fdd75c51511681e (259ms)
[warn] 	module not found: com.typesafe.akka#akka-http_2.11;10.1.1
[warn] ==== Maven2 Local: tried
[warn]   file:/home/travis/.m2/repository/com/typesafe/akka/akka-http_2.11/10.1.1/akka-http_2.11-10.1.1.pom
[warn] ==== jcenter: tried
[warn]   https://jcenter.bintray.com/com/typesafe/akka/akka-http_2.11/10.1.1/akka-http_2.11-10.1.1.pom
[warn] ==== jfrog: tried
[warn]   http://oss.jfrog.org/oss-snapshot-local/com/typesafe/akka/akka-http_2.11/10.1.1/akka-http_2.11-10.1.1.pom
[warn] 	::::::::::::::::::::::::::::::::::::::::::::::
[warn] 	::          UNRESOLVED DEPENDENCIES         ::
[warn] 	::::::::::::::::::::::::::::::::::::::::::::::
[warn] 	:: com.typesafe.akka#akka-http_2.11;10.1.1: not found
[warn] 	::::::::::::::::::::::::::::::::::::::::::::::
```

It looks like the pom.sha1 file is actually the pom. 

```
$ curl -Ls 'https://jcenter.bintray.com/com/typesafe/akka/akka-http_2.11/10.1.1/akka-http_2.11-10.1.1.pom.sha1' | head
<?xml version='1.0' encoding='UTF-8'?>
<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://maven.apache.org/POM/4.0.0">
    <modelVersion>4.0.0</modelVersion>
    <groupId>com.typesafe.akka</groupId>
    <artifactId>akka-http_2.11</artifactId>
    <packaging>jar</packaging>
    <description>akka-http</description>
    <url>https://akka.io</url>
    <version>10.1.1</version>
    <licenses>
```

It seems to be fine on maven central though:

```
$ curl -s 'https://repo1.maven.org/maven2/com/typesafe/akka/akka-http_2.11/10.1.1/akka-http_2.11-10.1.1.pom.sha1'
7ab0cc491f566d7824d76c3f7fdd75c51511681e
```

**Comment by ktoso** [comment]
The source of truth is maven central. So that's good.

bintray has been having terrible issues the entire week so all the publishes are wrong there for 10.1.1 -- http://status.bintray.com/ -- or rather, I have deleted them all instead of keeping wrong artifacts around; in fact, they should be deleted and not wrong; Depending on bintray for artifacts is like depending on "staging" in this case;

**Comment by ktoso** [comment]
In other words https://bintray.com/akka/maven/com.typesafe.akka:akka-http_2.11/10.1.1 this is expected not to exist right now.

We continue to work with bintray support to make sure bintray  ends up containing the files though. I'll keep this open for now

**Comment by 2m** [comment]
We should remove 10.1.1 from JCenter because that release is incomplete because of Bintray issues.

**Comment by ktoso** [comment]
Marking as bug, though "release process bug" is perhaps a bit weird but true in this case

**Comment by 2m** [comment]
10.1.1 was removed from JCenter. Are you only using JCenter resolver?

If you have Maven Central as well as JCenter in the resolvers then now Ivy should see that there is no trace of 10.1.1 in JCenter and will pick Maven Central instead.

**Comment by ktoso** [comment]
They have worked around it -- see the linked PR. It is as simple as making the maven one "first" in the resolvers. But still, we should get proper artifacts onto there I guess.

Or do we say that them being removed is fine for this one?
After all, bintray is more like a staging thing for us and should not be treated as source of truth by end users

**Comment by 2m** [comment]
It seems that they did not reorder, but added maven central instead. In that case we should put 10.1.1 onto JCenter when publishing is fixed on the Bintray side.

**Comment by brharrington** [comment]
We're not blocked and it was trivial to work around. It is just that it used to work. Maven central was likely removed by mistake at some point years ago in our build and since jcenter is typically a superset nobody noticed.

In this case, I think what was bad is that it was inconsistent. The 2.12 version resolved fine against jcenter where the 2.11 dependency did not. It was also odd to see that the expected sha1 value in the error was xml instead of a hex string. Normally with our projects I have seen failures with a subset of files failing to get pushed rather than a file having the wrong content.

Note, I still see some traces of 10.1.1 on jcenter, for example:

https://jcenter.bintray.com/com/typesafe/akka/akka-http_2.11/10.1.1/

**Comment by raboof** [comment]
I think we should not worry about this for 10.1.1 (and make sure 10.1.2 either gets there completely or not at all)
